disnake/channel.py

```diff
@@ -1817,6 +1853,473 @@ class NewsChannel(TextChannel):
+        "name",
+        "category_id",
+        "topic",
+        "position",
+        "nsfw",
+        "last_thread_id",
+        "default_auto_archive_duration",
+        "guild",
+        "_state",
+        "_type",
+        "_overwrites",
+    )
+
+    def __init__(self, *, state: ConnectionState, guild: Guild, data: ForumChannelPayload) -> None:
+        self._state: ConnectionState = state
+        self.id: int = int(data["id"])
+        self._type: int = data["type"]
+        self._update(guild, data)
+
+    def __repr__(self) -> str:
+        atts = [
+            ("id", self.id),
+            ("name", self.name),
+            ("topic", self.topic),
+            ("position", self.position),
+            ("nsfw", self.nsfw),
+            ("category_id", self.category_id),
+            ("default_auto_archive_duration", self.default_auto_archive_duration),
+        ]
+        joined = " ".join("%s=%r" % t for t in atts)
+        return f"<{type(self).__name__} {joined}>"
+
+    def _update(self, guild: Guild, data: ForumChannelPayload) -> None:
+        self.guild: Guild = guild
+        self.name: str = data["name"]
+        self.category_id: Optional[int] = utils._get_as_snowflake(data, "parent_id")
+        self.topic: Optional[str] = data.get("topic")
+        self.position: int = data["position"]
+        self.nsfw: bool = data.get("nsfw", False)
+        self.last_thread_id: Optional[int] = utils._get_as_snowflake(data, "last_message_id")
+        self.default_auto_archive_duration: ThreadArchiveDurationLiteral = data.get(
+            "default_auto_archive_duration", 1440
+        )
+        self._fill_overwrites(data)
+
+    async def _get_channel(self) -> ForumChannel:  # I wonder why this exists
+        return self
+
+    typing = disnake.abc.Messageable.typing
+    trigger_typing = disnake.abc.Messageable.trigger_typing
+
+    @property
+    def type(self) -> ChannelType:
+        return try_enum(ChannelType, self._type)
+
+    @property
+    def _sorting_bucket(self) -> int:
+        return ChannelType.text.value
+
+    @utils.copy_doc(disnake.abc.GuildChannel.permissions_for)
+    def permissions_for(
+        self,
+        obj: Union[Member, Role],
+        /,
+        *,
+        ignore_timeout: bool = MISSING,
+    ) -> Permissions:
+        base = super().permissions_for(obj, ignore_timeout=ignore_timeout)
+
+        # forum channels do not have voice related permissions
+        denied = Permissions.voice()
+        base.value &= ~denied.value
+        return base
+
+    @property
+    def threads(self) -> List[Thread]:
+        """List[:class:`Thread`]: Returns all the threads that you can see."""
+        return [thread for thread in self.guild._threads.values() if thread.parent_id == self.id]
+
+    def is_nsfw(self) -> bool:
+        """Whether the channel is marked as NSFW
+
+        :return type: :class:`bool`
+        """
+        return self.nsfw
+
+    @property
+    def last_thread(self) -> Optional[Thread]:
```

Comment:
Hm, I'm unsure about this. @onerandomusername what do you think?

Reply:
Let's keep last_thread but remove fetch_thread, as it is just a shortcut to the guild's method.

Reply:
Then again, get_thread exists on text channels, so I don't really know. Perhaps we should keep it, but it would be harder to remove.

Given the bugs existant with thread caching right now, we should probably not implement this method for now, as it would be easier to add it than to remove it.

Reply:
https://github.com/DisnakeDev/disnake/pull/448/commits/4de2079abe4602f9dd2df72168d0230a23a8de69